src/panels/config/entity_registry/ha-config-entity-registry.ts

```diff
@@ -192,8 +383,106 @@ class HaConfigEntityRegistry extends LitElement {
     }
   }
 
-  private _showDisabledChanged(ev: Event) {
-    this._showDisabled = (ev.target as HaSwitch).checked;
+  private _showDisabledChanged() {
+    this._showDisabled = !this._showDisabled;
+  }
+
+  private _showRestoredChanged() {
+    this._showRestored = !this._showRestored;
+  }
+
+  private _handleSearchChange(ev: CustomEvent) {
+    this._filter = ev.detail.value;
+  }
+
+  private _handleSelectionChanged(ev: CustomEvent): void {
+    const changedSelection = ev.detail as SelectionChangedEvent;
+    const entity = changedSelection.id;
+    if (changedSelection.selected) {
+      this._selectedEntities.push(entity);
+    } else {
+      const index = this._selectedEntities.indexOf(entity);
+      if (index !== -1) {
+        this._selectedEntities.splice(index, 1);
+      }
+    }
+    this._selectedEntities = [...this._selectedEntities];
```

Comment:
why are we making a copy here? shouldn't we make a copy earlier in this method when we're mutating it ?